### PR TITLE
Update k8s.io Go modules to v0.20.4

### PR DESCRIPTION
**Pull request by bmelbourne**
**What this PR does / why we need it**:
Update k8s.io Go modules to **v0.20.4**.

- k8s.io/api v0.20.0 => k8s.io/api v0.20.4
- k8s.io/apiextensions-apiserver v0.20.0 => k8s.io/apiextensions-apiserver v0.20.4
- k8s.io/apimachinery v0.20.0 => k8s.io/apimachinery v0.20.4
- k8s.io/apiserver v0.20.0 => k8s.io/apiserver v0.20.4
- k8s.io/cli-runtime v0.20.0 => k8s.io/cli-runtime v0.20.4
- k8s.io/client-go v0.20.0 => k8s.io/client-go v0.20.4
- k8s.io/cloud-provider v0.20.0 => k8s.io/cloud-provider v0.20.4
- k8s.io/cluster-bootstrap v0.20.0 => k8s.io/cluster-bootstrap v0.20.4
- k8s.io/component-base v0.20.0 => k8s.io/component-base v0.20.4
- k8s.io/component-helpers v0.20.0 => k8s.io/component-helpers v0.20.4
- k8s.io/controller-manager v0.20.0 => k8s.io/controller-manager v0.20.4
- k8s.io/cri-api v0.20.0 => k8s.io/cri-api v0.20.4
- k8s.io/csi-translation-lib v0.20.0 => k8s.io/csi-translation-lib v0.20.4
- k8s.io/kube-aggregator v0.20.0 => k8s.io/kube-aggregator v0.20.4
- k8s.io/kube-controller-manager v0.20.0 => k8s.io/kube-controller-manager v0.20.4
- k8s.io/kube-proxy v0.20.0 => k8s.io/kube-proxy v0.20.4
- k8s.io/kube-scheduler v0.20.0 => k8s.io/kube-scheduler v0.20.4
- k8s.io/kubectl v0.19.4 => k8s.io/kubectl v0.20.4
- k8s.io/kubelet v0.20.0 => k8s.io/kubelet v0.20.4
- k8s.io/kubernetes v0.20.0 => k8s.io/kubernetes v1.20.4
- k8s.io/legacy-cloud-providers v0.20.0 => k8s.io/legacy-cloud-providers v0.20.4
- k8s.io/metrics v0.20.0 => k8s.io/metrics v0.20.4
- k8s.io/mount-utils v0.20.0 => k8s.io/mount-utils v0.20.4
- k8s.io/sample-apiserver v0.20.0 => k8s.io/sample-apiserver v0.20.4
- k8s.io/code-generator v0.20.0 => k8s.io/code-generator v0.20.4
- github.com/docker/docker v1.4.2-0.20200309214505-aa6a9891b09c => github.com/docker/docker v17.12.0-ce-rc1.0.20200916142827-bd33bbf0497b+incompatible

**Which issue(s) this PR fixes**:
N/A

**Special notes for your reviewer**:
None

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -2,34 +2,37 @@ module k8s.io/kops
 
 go 1.14
 
-// Version kubernetes-1.20.0-beta.2 => tag v0.20.0
+// Version kubernetes-1.20.4 => tag v0.20.4
```
Comment (hakman) [inline go.mod:5]: Please remove this comment.

File: go.mod [hunk 2]
```diff
@@ -2,34 +2,37 @@ module k8s.io/kops
...
+	k8s.io/kubernetes => k8s.io/kubernetes v1.20.4
+	k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.20.4
+	k8s.io/metrics => k8s.io/metrics v0.20.4
+	k8s.io/mount-utils => k8s.io/mount-utils v0.20.4
+	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.20.4
 )
 
 // This should match hack/go.mod
-replace k8s.io/code-generator => k8s.io/code-generator v0.20.0
+replace k8s.io/code-generator => k8s.io/code-generator v0.20.4
```
Comment (hakman) [inline go.mod:35]: Please move this to the top, to be very visible (see my other PR)

File: go.mod (hunk 1 above)
Comment (olemarkus) [reply go.mod:5]: Don't have to update it at least. The comment is nice enough as I always forget to remove the 1.

Comment (hakman) [reply go.mod:5]: Easy to see that it break things later :)

Comment (bmelbourne) [reply go.mod:5]: @hakman 
Are you happy for me to leave it in?
I agree with @olemarkus that it's good to see how most of the `k8s.io` Go modules versioning relates to the release versioning of Kubernetes. It certainly helped me when I started working on the recent PRs.

Comment (hakman) [reply go.mod:5]: I think so, just please move the other on the top.

**Comment by rifelpet** [comment]
/lgtm
I'll defer to @hakman for final approval

**Comment by hakman** [comment]
Thanks @rifelpet 
/approve

**Pull request merged**